src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,54 @@
+package core.basesyntax;
+
+public class IsoscelesTrapezoid extends Figure {
+    private double topBase;
+    private double bottomBase;
+    private double sideBase;
+
+    public IsoscelesTrapezoid(double topBase, double bottomBase, double sideBase, Colour colour) {
+        super(colour);
+        this.topBase = topBase;
+        this.bottomBase = bottomBase;
+        this.sideBase = sideBase;
+
+    }
+
+    public double getHigh() {
+        return 0.5 * Math.sqrt(4 * sideBase * sideBase - (Math.pow(topBase - bottomBase, 2)));
+    }
+
+    @Override
+    public double getArea() {
+        return getHigh() / 2 * (topBase + bottomBase);
+    }
+
+    @Override
+    public double getPerimeter() {
+        return sideBase * 2 + topBase + bottomBase;
+    }
+
+    public double getTopBase() {
+        return topBase;
+    }
+
+    public double getBottomBase() {
+        return bottomBase;
+    }
+
+    public double getSideBase() {
+        return sideBase;
+    }
+
+    @Override
+    public String draw() {
+        return "IsoscelesTrapezoid:"
+                + " high: " + parsingValue(getHigh())
+                + ", perimeter" + parsingValue(getPerimeter()) + " units"
+                + ", area: " + parsingValue(getArea()) + " units"
+                + ", color: " + getColour()
+                + ", area: " + parsingValue(getArea()) + " units"
+                + ", base top:" + parsingValue(topBase) + " units"
+                + ", base bot:" + parsingValue(bottomBase) + " units"
+                + ", base side :" + parsingValue(sideBase) + " units";
```

Comment:
Наприклад це метод може виглядати так:
```suggestion
        return String.format("IsoscelesTrapezoid: high: %.2f units, perimeter: %.2f units,"
                + " area: %.2f units, color: %s, base top: %.2f units," 
                + " base bot: %.2f units, base side: %.2f usnits",
                getHigh(), getPerimeter(), getArea(), getColour(), 
                topBase, bottomBase, sideBase);
```

Reply:
_Надеюсь ты прочтешь этот комментарий завтра. А то как стыдно, что ты смотришь на код почти в 9 вечера._
Не совсем согласен, с точки зрения общей абстракции.
Мне чтоб, к примеру, изменить с 2 цифр на 5 (после запятой) нужно заменить всего одну строчку. 
Если, все таки стоит сделать по твоему совету - просто поставь плюс, пожалуйста.

Reply:
Я поспал с этой мыслью, беру свои слова назад.